### PR TITLE
App Power Forms connector artifacts for certification

### PR DESCRIPTION
**Pull request by TerhoAntila**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Comment by natalie-pienkowska** [comment]
Hi @TerhoAntila! Our certification engineer, Srikanth, is currently on emergency leave. Please expect a delay with your PR. He will return January 31st, 2022. We apologize for any inconveniences.

**Comment by TerhoAntila** [comment]
> Hi @TerhoAntila! Our certification engineer, Srikanth, is currently on emergency leave. Please expect a delay with your PR. He will return January 31st, 2022. We apologize for any inconveniences.

Thanks for the info. I was also contacted two days ago via email by Amjed Ayoub. I just can't get over the fact that the whole certification process is so drastically dependent on a single person.

> Hi @TerhoAntila! Our certification engineer, Srikanth, is currently on emergency leave. Please expect a delay with your PR. He will return January 31st, 2022. We apologize for any inconveniences.

Thanks for the info. Hopefully everything will turn out for the best with Srikanth and/or his close ones.

File: certified-connectors/App Power Forms/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,647 @@
...
+    "title": "App Power Forms",
+    "description": "Use this connector from App Power LLC to easily include external stakeholders in your business processes. The connector makes it very easy to create a form within the designer. You can start with a free trial license and acquire a production license at https://apppower.net.",
+    "version": "1.0",
+    "contact": {
+      "name": "App Power",
+      "url": "https://apppower.net",
+      "email": "support@apppower.net"
+    }    
+  },
+  "host": "apppower-forms-dev.azurewebsites.net",
```
Comment (sriyen-msft) [inline certified-connectors/App Power Forms/apiDefinition.swagger.json:13]: @TerhoAntila We do not certify the connectors having dev/staging/test host URL. Please provide a production version of host to move forward for certification.

File: certified-connectors/App Power Forms/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,647 @@
...
+        ],
+        "description": "Use this action to create a form based on Adaptive Card JSON."
+      }
+    },
+    "/AddCardSchema/{cardName}": {
+      "post": {
+        "responses": {
+          "200": {
+            "description": "No need to specify response schema",
+            "schema":{}
```
Comment (sriyen-msft) [inline certified-connectors/App Power Forms/apiDefinition.swagger.json:265]: @TerhoAntila Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

File: certified-connectors/App Power Forms/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,647 @@
...
+          }
+        },
+        "operationId": "AddCardSchema",
+        "x-ms-visibility": "internal",
+        "parameters": [
+          {
+            "name": "cardData",
+            "in": "body",
+            "required": false,
+            "schema": {}
```
Comment (sriyen-msft) [inline certified-connectors/App Power Forms/apiDefinition.swagger.json:275]: @TerhoAntila Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

File: certified-connectors/App Power Forms/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,647 @@
...
+          }
+        ]
+      }
+    },
+    "/AddCardFieldSchema/{formName}/{fieldName}": {
+      "post": {
+        "responses": {
+          "200": {
+            "description": "No need to specify response schema",
+            "schema":{}
```
Comment (sriyen-msft) [inline certified-connectors/App Power Forms/apiDefinition.swagger.json:292]: @TerhoAntila Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

File: certified-connectors/App Power Forms/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,647 @@
...
+          }
+        },
+        "operationId": "AddCardFieldSchema",
+        "x-ms-visibility": "internal",
+        "parameters": [
+          {
+            "name": "fieldType",
+            "in": "body",
+            "required": false,
+            "schema": {}
```
Comment (sriyen-msft) [inline certified-connectors/App Power Forms/apiDefinition.swagger.json:302]: @TerhoAntila Empty property values can be removed until unless you have a specific reason to keep them.

File: certified-connectors/App Power Forms/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,647 @@
...
+          }
+        ]
+      }
+    },
+    "/GetResponseSchema/{cardName}": {
+      "get": {
+        "responses": {
+          "200": {
+            "description": "No need to specify response schema",
+            "schema":{}
```
Comment (sriyen-msft) [inline certified-connectors/App Power Forms/apiDefinition.swagger.json:385]: @TerhoAntila Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

File: certified-connectors/App Power Forms/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,647 @@
...
+        "operationId": "MetadataFormFieldTypes",
+        "x-ms-visibility": "internal"
+      }
+    },
+    "/MetadataFieldTypeSpecificSchema": {
+      "post": {
+        "responses": {
+          "200": {
+            "description": "No need",
+            "schema":{}
```
Comment (sriyen-msft) [inline certified-connectors/App Power Forms/apiDefinition.swagger.json:457]: @TerhoAntila Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

File: certified-connectors/App Power Forms/apiDefinition.swagger.json [hunk 8]
```diff
@@ -0,0 +1,647 @@
...
+          }
+        },
+        "operationId": "MetadataFieldTypeSpecificSchema",
+        "x-ms-visibility": "internal",
+        "parameters": [
+          {
+            "name": "fieldType",
+            "in": "body",
+            "required": false,
+            "schema": {}
```
Comment (sriyen-msft) [inline certified-connectors/App Power Forms/apiDefinition.swagger.json:467]: @TerhoAntila Empty property values can be removed until unless you have a specific reason to keep them.

File: certified-connectors/App Power Forms/apiProperties.json [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+          }
+        }
+      }
+    },
+    "iconBrandColor": "#6DD47E",
+    "capabilities": [], 
+    "policyTemplateInstances": [
+      {
+        "templateId": "pollingtrigger",
+        "title": "PollingTriggerPolicy_69bf0411-f6e0-cc7e-e7d1-5b504c8e9669",
```
Comment (sriyen-msft) [inline certified-connectors/App Power Forms/apiProperties.json:23]: @TerhoAntila No secret-look-a-like values are allowed in the swagger due to Microsoft security policy.

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

File: certified-connectors/App Power Forms/apiDefinition.swagger.json (hunk 2 above)
Comment (TerhoAntila) [reply certified-connectors/App Power Forms/apiDefinition.swagger.json:265]: @sriyen-msft Could you please clarify this. This and all similar findings concern an internal operation and are used by x-ms-dynamic-schema definition. Surely x-ms-dynamic-schema is supported? The makers do NOT need to use parse JSON to parse any of the responses, they will be able to use dynamic content picker (please see the attached image)
image
This comment applies to all findings with similar description.

File: certified-connectors/App Power Forms/apiDefinition.swagger.json (hunk 3 above)
Comment (TerhoAntila) [reply certified-connectors/App Power Forms/apiDefinition.swagger.json:275]: No parse JSON required.
image

File: certified-connectors/App Power Forms/apiDefinition.swagger.json (hunk 6 above)
Comment (TerhoAntila) [reply certified-connectors/App Power Forms/apiDefinition.swagger.json:385]: No parse JSON required.
image

File: certified-connectors/App Power Forms/apiDefinition.swagger.json (hunk 7 above)
Comment (TerhoAntila) [reply certified-connectors/App Power Forms/apiDefinition.swagger.json:457]: No parse JSON required here either. This operation is used to display different set of fields based on the field type selection made by the user. See the attached pic.
image

File: certified-connectors/App Power Forms/apiProperties.json (hunk 9 above)
Comment (TerhoAntila) [reply certified-connectors/App Power Forms/apiProperties.json:23]: Will fix...

File: certified-connectors/App Power Forms/apiDefinition.swagger.json (hunk 1 above)
Comment (TerhoAntila) [reply certified-connectors/App Power Forms/apiDefinition.swagger.json:13]: How did that end up there 😅 Will fix...

File: certified-connectors/App Power Forms/apiDefinition.swagger.json (hunk 2 above)
Comment (sriyen-msft) [reply certified-connectors/App Power Forms/apiDefinition.swagger.json:265]: @TerhoAntila The "AddCard" operation might receive a response with a "Success" or "Failed" message which can be defined in the schema, like "type": "string". Empty schema definitions are not allowed anymore.

Comment (sriyen-msft) [reply certified-connectors/App Power Forms/apiDefinition.swagger.json:265]: Applies the same comment to all the similar questions.

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

File: certified-connectors/App Power Forms/apiDefinition.swagger.json (hunk 2 above)
Comment (TerhoAntila) [reply certified-connectors/App Power Forms/apiDefinition.swagger.json]: @sriyen-msft Just made a commit resolving empty schema definitions.

File: certified-connectors/App Power Forms/apiDefinition.swagger.json (hunk 3 above)
Comment (TerhoAntila) [reply certified-connectors/App Power Forms/apiDefinition.swagger.json:277]: Resolution committed.

File: certified-connectors/App Power Forms/apiDefinition.swagger.json (hunk 4 above)
Comment (TerhoAntila) [reply certified-connectors/App Power Forms/apiDefinition.swagger.json]: Resolution committed.

File: certified-connectors/App Power Forms/apiDefinition.swagger.json (hunk 6 above)
Comment (TerhoAntila) [reply certified-connectors/App Power Forms/apiDefinition.swagger.json]: Resolution committed.

File: certified-connectors/App Power Forms/apiDefinition.swagger.json (hunk 7 above)
Comment (TerhoAntila) [reply certified-connectors/App Power Forms/apiDefinition.swagger.json]: Resolution committed.

File: certified-connectors/App Power Forms/apiProperties.json (hunk 9 above)
Comment (TerhoAntila) [reply certified-connectors/App Power Forms/apiProperties.json]: Resolution committed.

File: certified-connectors/App Power Forms/apiDefinition.swagger.json (hunk 5 above)
Comment (TerhoAntila) [reply certified-connectors/App Power Forms/apiDefinition.swagger.json:306]: This is needed.

File: certified-connectors/App Power Forms/apiDefinition.swagger.json (hunk 8 above)
Comment (TerhoAntila) [reply certified-connectors/App Power Forms/apiDefinition.swagger.json:475]: This is needed

**Review by sriyen-msft — APPROVED** [review]

**Pull request merged**